src/handlers/InputAdded.ts

```diff
@@ -82,12 +144,23 @@ export default class InputAdded implements Handler {
                 blockHash: log.block.hash,
                 transactionHash: log.transaction?.hash,
             });
+
             const erc20Deposit = await this.handlePayload(input, block, ctx);
             if (erc20Deposit) {
                 this.depositStorage.set(inputId, erc20Deposit);
                 ctx.log.info(`${inputId} (Erc20Deposit) stored`);
                 input.erc20Deposit = erc20Deposit;
             }
+
+            input.erc721Deposit = await this.prepareErc721Deposit(
```

Comment:
this block of code stayed a little bit different from the token handling above. Is there a way to make it more uniform?

Reply:
I think the `handlePayload()` deserves a bit of refactoring following the steps of `prepareErc721Deposit` in a separate PR. When I started, I did add the logic inside it. I would return an object with a return like `{erc721Deposit: ERC20 } | {erc20Deposit: NFT}` so it would be mutually exclusive because the `msgSender` is 1:1.  Also, I was not fond of the duplication on checking if it exists than store/log/assign. I prefer to separate and have a method to take care of everything at once and leave the `handle()` to control and glue for the `Input`. That is for the context where I am coming from. I can create an issue and refactor that in a separate branch.